src/chainstate/stacks/boot/contract_tests.rs

```diff
@@ -429,6 +432,908 @@ impl HeadersDB for TestSimHeadersDB {
     }
 }
 
+#[test]
+fn pox_2_contract_caller_units() {
+    let mut sim = ClarityTestSim::new();
+    sim.epoch_bounds = vec![0, 1];
+    let delegator = StacksPrivateKey::new();
+
+    // execute past 2.1 epoch initialization
+    sim.execute_next_block(|_env| {});
+    sim.execute_next_block(|_env| {});
+    sim.execute_next_block(|_env| {});
+
+    sim.execute_next_block(|env| {
+        env.initialize_contract(POX_2_CONTRACT_TESTNET.clone(), &POX_2_TESTNET_CODE, None)
+            .unwrap()
+    });
+
+    let cc = boot_code_id("stack-through", false);
+
+    sim.execute_next_block(|env| {
+        env.initialize_contract(cc.clone(),
+                                "(define-public (cc-stack-stx (amount-ustx uint)
+                                                           (pox-addr (tuple (version (buff 1)) (hashbytes (buff 20))))
+                                                           (start-burn-ht uint)
+                                                           (lock-period uint))
+                                   (contract-call? .pox-2 stack-stx amount-ustx pox-addr start-burn-ht lock-period))",
+                                None)
+            .unwrap();
+
+        let burn_height = env.eval_raw("burn-block-height").unwrap().0;
+
+        assert_eq!(
```

Comment:
can you put a brief comment for each of these cases to explain why we get these results? because I think it's not obvious.

Reply:
Added in 199ce4126